src/Raven.Server/ServerWide/Maintenance/ClusterMaintenanceSupervisor.cs

```diff
@@ -248,7 +249,16 @@ private void ListenToMaintenanceWorker()
 
                                 var report = BuildReport(rawReport);
                                 timeoutEvent.Defer(_parent._leaderClusterTag);
-
+                                foreach (var name in report.Report.Keys.ToList())
```

Comment:
Better to avoid this allocation (should be false most of the time when we are active) and only allocate if we indeed get a no chance notification.
For that matter, since we are single threaded, we can just have a separate list of the modified values that we'll then set after the `foreach`

Reply:
Fixed.